pkg/controller/cainjector/indexers.go

```diff
@@ -35,11 +35,11 @@ import (
 
 // certificateToInjectableFunc converts a given certificate to the reconcile requests for the corresponding injectables
 // (webhooks, api services, etc) that reference it.
-type certificateToInjectableFunc func(log logr.Logger, cl client.Client, certName types.NamespacedName) []ctrl.Request
+type certificateToInjectableFunc func(log logr.Logger, cl client.Reader, certName types.NamespacedName) []ctrl.Request
```

Comment:
These functions only ever read from the API and should always be reading from a cache, so I have made that explicit by expecting a Reader here and supplying a `cache.Cache` as the client.